boards/kernel_layout.ld

```diff
@@ -218,6 +230,25 @@ SECTIONS
     {
         . = ALIGN(4);
         _srelocate = .;
+
+        /* The Global Pointer is used by the RISC-V architecture to provide
```

Comment:
Data past 4kB has to do something else (that's less efficient)? It seems like there are certain kernel structures we'll want to make sure are in this 4kB region, then.